sdk/storage/Azure.Storage.Blobs/api/Azure.Storage.Blobs.netstandard2.0.cs

```diff
@@ -1405,6 +1410,26 @@ public partial class BlobBaseClient
         protected virtual Azure.Storage.Blobs.Specialized.BlobBaseClient WithSnapshotCore(string snapshot) { throw null; }
         public virtual Azure.Storage.Blobs.Specialized.BlobBaseClient WithVersion(string versionId) { throw null; }
     }
+    public partial class BlobDirectoryClient
+    {
+        protected BlobDirectoryClient() { }
+        public BlobDirectoryClient(string connectionString, string blobContainerName, string blobDirectoryPath) { }
+        public BlobDirectoryClient(string connectionString, string blobContainerName, string blobDirectoryPath, Azure.Storage.Blobs.BlobClientOptions options) { }
+        public BlobDirectoryClient(System.Uri blobUri, Azure.AzureSasCredential credential, Azure.Storage.Blobs.BlobClientOptions options = null) { }
+        public BlobDirectoryClient(System.Uri blobUri, Azure.Core.TokenCredential credential, Azure.Storage.Blobs.BlobClientOptions options = null) { }
+        public BlobDirectoryClient(System.Uri blobUri, Azure.Storage.Blobs.BlobClientOptions options = null) { }
+        public BlobDirectoryClient(System.Uri blobUri, Azure.Storage.StorageSharedKeyCredential credential, Azure.Storage.Blobs.BlobClientOptions options = null) { }
+        public virtual System.Uri Uri { get { throw null; } }
+        public Azure.Storage.Blobs.Specialized.BlobDirectoryClient CloneClient() { throw null; }
+        public virtual Azure.Response Download(string path) { throw null; }
+        public virtual Azure.Response Download(string path, Azure.Storage.Blobs.Models.BlobRequestConditions conditions = null, Azure.Storage.StorageTransferOptions transferOptions = default(Azure.Storage.StorageTransferOptions), System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```

Comment:
I'm not sure we should take request conditions. is it going to be applied to all blobs in path ?

Reply:
Is the reason because customers may run into user error like setting the `IfMatch` to a single Etag and `TagConditions`? I think some customers may want to use `IfModifiedSince`, `IfUnmodifiedSince`, so should I create a separate RequestionConditions for this or just omit the option of RequestConditions all together?

Reply:
I think we can omit RequestConditions for now to reduce the scope. When we add this back (or keep) I think we should consider separate class for directory as a lot of properties in RequestCondidtions wouldn't work or had weird semantics.
Also if modified/unmodified are the only conditions that we think would make sense maybe we should have DirectoryDownloadOptions and put them there without any nesting.